x/stakeibc/keeper/lsm_token_deposit.go

```diff
@@ -0,0 +1,51 @@
+package keeper
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/Stride-Labs/stride/v8/x/stakeibc/types"
+)
+
+func (k Keeper) SetLSMTokenDeposit(ctx sdk.Context, deposit types.LSMTokenDeposit) {
+	// TODO [LSM]
+}
+
+func (k Keeper) RemoveLSMTokenDeposit(ctx sdk.Context, denom string) {
+	// TODO [LSM]
+}
+
+func (k Keeper) GetLSMTokenDeposit(ctx sdk.Context, denom string) (deposit types.LSMTokenDeposit, found bool) {
```

Comment:
For Remove and Get don't we also need the validator information to arrive at a single specific deposit?  The prefix store will use chain_id then validator then denom as key prefixes, the CTX can tell us the chain_id but can it also tell us the validator or should this also be an input (or return multiple deposits across all validators which match this denom)?

Reply:
Good call out! I think we have two options here:

1. Take three parameters, `chainId, validatorAddress, denom`
2. Take an ID field (that represents the concatenation of the fields)

I don't think this function has a clear use case yet since we typically access all the records from a chain. So I don't have a strong opinion. Maybe we start with option 2 and we can adjust later if needed? Wdyt?

Reply:
As I have continued to think about it more, the denom contains information about the validator (and implicitly the chain_id) already.  You would never be able to have the same denom present on more than one validator.  Handing in a single field like denom would force and iteration over the prefix store whereas all 3 params would allow immediate lookup but this small efficiency different probably doesn't matter.  It is making me wonder if the validator address should be part of the prefix at all (if the prefix store was just chain_id and then denom, you can lookup a specific deposit and get the validator address from the deposit without it being a key in the prefix store)

Reply:
that's a good point - I think we can remove the validator from the prefix. The only purpose before was when we had to look up all the deposits associated with a given validator, but that's also no longer relevant now that we're not hoarding

Reply:
chatted offline - agreed that validator field is not necessary, but we still need chainId for efficient lookups